### PR TITLE
Side Cart On Grant List Screen

### LINKED ISSUE #14: Cart Functionality On Grant List Page
- [x] When a user clicks "Add To Cart" put the grant into the `localStorage` cart
- [x] Open up the side cart

### PR DESCRIPTION
**Pull request by apbendi**
Adds side cart functionality to the Grant Details page. Currently, scrolling on this page is an issue. Fix TBD.

closes #14 

Also populates tokens based on the global list, filtering based on the token the grant accepts if not all tokens. Adds the grant's accepted token to the cart data.

### CODE REVIEW

File: app/grants/templates/grants/detail/index.html [hunk 1]
```diff
@@ -67,7 +68,7 @@
       <div id="side-cart" class="grant-side-cart px-5 px-md-4 pb-5 pb-md-0 pt-md-0" style="display: none;">
         {% include 'grants/detail/side-cart.html' %}
       </div>
-      </div>
+      </div> <!-- meh? -->
```
Comment (mds1) [inline app/grants/templates/grants/detail/index.html:71]: What was this comment for? 😁

File: app/assets/v2/css/kudos/styles.css [hunk 2]
```diff
@@ -615,9 +615,9 @@ a.cellarius-partner:before {
   outline: none;
   background-color: transparent;
   display: flex; }
-.flip-card:focus .flip-card__flipper {
+/* .flip-card:focus .flip-card__flipper {
```
Comment (mds1) [inline app/assets/v2/css/kudos/styles.css:618]: Is this the wrong flip-card class? Two reasons I'm thinking this:

1. It's located in the `kudos` folder, as opposed to the `grants` folder
2. When on the grants listing page (http://0.0.0.0:8000/grants/?network=rinkeby&category=), clicking the "Add to Cart" button adds the grant to the cart and leaves the card flipped, as shown below. Can we get this to not flip at all when clicking Add to Cart?

image

File: app/assets/v2/css/kudos/styles.css [hunk 3]
```diff
@@ -635,9 +635,9 @@ a.cellarius-partner:before {
   border-radius: .25rem;
   z-index: 2;
   transform: rotateY(0); }
-.flip-card__flipper:hover {
+/* .flip-card__flipper:hover {
```
Comment (mds1) [inline app/assets/v2/css/kudos/styles.css:638]: Same comment about flip-card class as above

File: app/assets/v2/css/kudos/styles.css (hunk 2 above)
Comment (apbendi) [reply app/assets/v2/css/kudos/styles.css]: So this is weird, I think if you do a hard refresh, you'll see it doesn't refresh, but I think you're right that this CSS might be used for kudos as well. Uncommented the classes and simply removed them from the grant details div.

File: app/grants/templates/grants/detail/index.html (hunk 1 above)
Comment (apbendi) [reply app/grants/templates/grants/detail/index.html]: Good catch! Noticed this extra `div` tag at some point and wanted to double check it, then forgot. It's not needed & is now removed.

**Comment by mds1** [comment]
One last question and then it's good to merge, as long as you don't have anything else you wanted to do on this branch. 

What is commit 878be57 for? The commit message is "Don't show Add To Cart for team grants on list page". I'm only looking at the three default grants on http://0.0.0.0:8000/grants/?network=rinkeby&category=, and I'm not sure exactly what this commit changes

**Comment by apbendi** [comment]
If the logged in user is a team member of the grant shown, it will not show
the add to cart button on that card in the grant list.

**Comment by mds1** [comment]
Ahh ok, it sounded like there was some other grant type called "team grants". Makes sense!

**Pull request merged**